mountpoint-s3/src/inode.rs

```diff
@@ -627,18 +742,41 @@ impl ReaddirHandle {
             }
         }
 
-        Ok(self.results.write().unwrap().pop_front())
+        Ok(self.compare_and_get_next())
     }
 
     /// Re-add an entry to the front of the queue if the consumer wasn't able to use it
     pub fn readd(&self, entry: LookedUp) {
-        self.results.write().unwrap().push_front(entry);
+        self.remote_results.write().unwrap().push_front(entry);
     }
 
     pub fn parent(&self) -> InodeNo {
         self.parent_ino
     }
 
+    fn compare_and_get_next(&self) -> Option<LookedUp> {
+        let mut local_locked = self.local_results.write().unwrap();
+        let mut remote_locked = self.remote_results.write().unwrap();
+        match (local_locked.front(), remote_locked.front()) {
+            (None, None) => None,
+            (None, Some(_)) => remote_locked.pop_front(),
+            (Some(_), None) => local_locked.pop_front(),
+            (Some(local_lookup), Some(remote_lookup)) => {
+                let ordering = local_lookup.inode.name().cmp(remote_lookup.inode.name());
```

Comment:
nit: lift the comment up here about how we need to consider the case where both inodes have the same name, and how we resolve that tie

Reply:
Done.